File: HRD_exam1/src/main/webapp/join.js

function join(){
	console.log('join')
	
	//유효성 검사
	if(document.joinform.custno.value == ""){
		alert('회원번호를 입력해주세요.')
		document.joinform.custno.focus(); //커서 이동
		return false;
	}
	
	if(document.joinform.custname.value == ""){
		alert('회원성명을 입력해주세요.')
		document.joinform.custname.focus(); //커서 이동
		return false;
	}
	
	if(document.joinform.phone.value == ""){
		alert('전화번호를 입력해주세요.')
		document.joinform.phone.focus(); //커서 이동
		return false;
	}
	
	if(document.joinform.address.value == ""){
		alert('회원주수롤 입력해주세요.')
		document.joinform.address.focus(); //커서 이동
		return false;
	}
	
	if(document.joinform.joindate.value == ""){
		alert('가입일자를 입력해주세요.')
		document.joinform.joindate.focus(); //커서 이동
		return false;
	}
	
	if(document.joinform.grade.value == ""){
		alert('고객등급 입력해주세요.')
		document.joinform.grade.focus(); //커서 이동
		return false;
	}
	if(document.joinform.city.value == ""){
		alert('도시코드 입력해주세요.')
		document.joinform.city.focus(); //커서 이동
		return false;
	}
	
	document.joinform.submit(); //폼 전송
	
	alert('회원등록이 완료되었습니다.')
	
	
}

File: HRD_exam1/src/main/java/controller/Join.java
package controller;

import java.io.IOException;
import java.util.ArrayList;

import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;


@WebServlet("/join")
public class Join extends HttpServlet {
	private static final long serialVersionUID = 1L;
       
   
    public Join() {
        
    }


	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		request.setCharacterEncoding("UTF-8");
		
		int custno = Integer.parseInt(request.getParameter("custno"));
		
		String custname = request.getParameter("custname");

		String phone = request.getParameter("phone");

		String address = request.getParameter("address");

		
		String joindate = request.getParameter("joindate");

		String grade = request.getParameter("grade");
		int city = Integer.parseInt(request.getParameter("city"));
		
		memberDto dto = new memberDto(custno, custname, phone, address, joindate, grade, city);
		
		
		Dao dao = new Dao();
		
		boolean result = dao.setMember(dto);
		
		if(result) {
			response.setCharacterEncoding("UTF-8");
			response.sendRedirect("/HRD_exam1/view.jsp");
		}
	}


	protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
		
		request.setCharacterEncoding("UTF-8");
		
		String custname = request.getParameter("custname");

		String phone = request.getParameter("phone");

		String address = request.getParameter("address");

		
		String joindate = request.getParameter("joindate");

		String grade = request.getParameter("grade");

		
		int city = Integer.parseInt(request.getParameter("city"));
		
		memberDto dto = new memberDto(custname, phone, address, joindate, grade, city);

		
		Dao dao = new Dao();
		boolean result = dao.join(dto);
		
		if(result) {
			response.setCharacterEncoding("UTF-8");
			response.sendRedirect("/HRD_exam1/join.jsp");
		}
	}


}
